commcare_connect/opportunity/visit_import.py

```diff
@@ -311,3 +313,133 @@ def get_status_by_completed_work_id(dataset):
+
+    latitude_index = _get_header_index(headers, LATITUDE_COL)
+    longitude_index = _get_header_index(headers, LONGITUDE_COL)
+    active_index = _get_header_index(headers, ACTIVE_COL)
+    radius_index = _get_header_index(headers, RADIUS_COL)
+    area_name_index = _get_header_index(headers, AREA_NAME_COL)
+
+    with transaction.atomic():
+        to_create = []
+        to_update = []
+        invalid_rows = []
+        seen_catchments = set()
+        missing_catchments = 0
+
+        opportunity_access = {}
+        username_index = None
+        if USERNAME_COL in headers:
+            username_index = _get_header_index(headers, USERNAME_COL)
+            opportunity_access = {
+                oa.user.username: oa
+                for oa in OpportunityAccess.objects.filter(opportunity=opportunity).select_related("user")
+            }
+
+        for row in dataset:
+            row = list(row)
+            try:
+                latitude = Decimal(row[latitude_index])
+                longitude = Decimal(row[longitude_index])
+                radius = int(row[radius_index])
+                active = row[active_index].lower().strip() == "yes"
+                area_name = str(row[area_name_index])
+
+                if latitude < Decimal("-90") or latitude > Decimal("90"):
+                    raise ValueError("Latitude must be between -90 and 90 degrees")
+
+                if longitude < Decimal("-180") or longitude > Decimal("180"):
+                    raise ValueError("Longitude must be between -180 and 180 degrees")
+
+                if not radius:
+                    raise ValueError("Radius must be an integer")
+
+                if active_index is None or row[active_index].lower().strip() not in ["yes", "no"]:
+                    raise ValueError("Active status must be 'yes' or 'no'")
+
+                if area_name_index is None or not isinstance(row[area_name_index], str):
+                    raise ValueError("Area name is not valid.")
+
+                if username_index is None or not row[username_index]:
+                    catchment = CatchmentArea(
+                        latitude=latitude,
+                        longitude=longitude,
+                        radius=radius,
+                        opportunity=opportunity,
+                        name=area_name,
+                        active=active,
+                    )
+                    missing_catchments += 1
+                    to_create.append(catchment)
+                elif row[username_index] in opportunity_access:
+                    username = row[username_index]
+                    catchment = None
+                    created = False
+                    if ID in headers and row[_get_header_index(headers, ID)]:
+                        catchment_id = row[_get_header_index(headers, ID)]
+                        catchment, created = CatchmentArea.objects.get_or_create(
```

Comment:
Why not just get it and if it's not there add to `to_create` so that it can take advantage of bulk creation?

Reply:
@sravfeyn Update the logic as per your suggestions.